pkg/storegateway/bucket.go

```diff
@@ -330,6 +330,9 @@ func (s *BucketStore) syncBlocks(ctx context.Context) error {
 		}
 		select {
 		case <-ctx.Done():
+			close(blockc)
+			wg.Wait()
+			return nil
```

Comment:
_Just to double-check, to get the idea of the problem here_ 

You mentioned,

> previously `BucketStore` would not wait for `syncBlocks` to finish before returning

Was this about this chunk? The new version looks like it does the same as what the code right below the `range metas` loop does. So the change here is to bail faster. But the previous code did wait, didn't it?

Reply:
actually yeah, you're right. We don't need this change
```suggestion
```